File: AtelieConfeiteira/ScheduleCake/agendabolo.API/agendabolo.domain/Core/Ingredientes/Ingrediente.cs
using Agendabolo.Core.Fabricantes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Agendabolo.Core.Ingredientes
{
    [Table("ingredientes")]
    public class Ingrediente
    {
        [Key]
        [Column("id")]
        public ulong Id { get; set; }

        [Column("nome")]
        public string Nome { get; set; }

        [Column("precocusto")]
        public decimal PrecoCusto { get; set; }

        [Column("quantidadeEmbalagem")]
        public decimal QuantidadeEmbalagem { get; set; }

        [Column("status")]
        public StatusCadastro Status { get; set; } = StatusCadastro.Normal;

        [JsonIgnore]
        public ICollection<Receitas.ReceitaIngrediente> Receitas { get; set; }
    }
}

File: AtelieConfeiteira/ScheduleCake/agendabolo.web/src/core/Ingredientes/Domain/Ingrediente.js
export default class Ingrediente {
  constructor(id, nome, precoCusto, quantidadeEmbalagem, status) {
    this.id = id;
    this.nome = nome;
    this.precoCusto = precoCusto;
    this.quantidadeEmbalagem = quantidadeEmbalagem;
    this.status = status;
  }

  id = 0;
  nome = '';
  precoCusto = 0;
  status = 0;
  quantidadeEmbalagem = 0;
}
